### PR TITLE
cli: add field filters to the CLI and tetragon configmap

### PR DESCRIPTION
**Pull request by willfindlay**
Implement field filters for the tetragon CLI and Tetragon ConfigMap for the JSON exporter. See individual commits.

### CODE REVIEW

**Review by michi-covalent — APPROVED** [review]
it's a technology

**Review by michi-covalent — APPROVED** [review]
technology

File: install/kubernetes/values.yaml [hunk 1]
```diff
@@ -89,6 +89,22 @@ tetragon:
...
+  # field paths can be separated by commas like "a.b.c,d,e.f". An optional "event_set" may
+  # be specified to apply the field filter to a specific set of events.
+  #
+  # For example, to exclude the "parent" field from all events and include the "process"
+  # field in PROCESS_KPROBE events while excluding all others:
+  #
+  # fieldFilters: |
+  #   {"fields": "parent", "action": "EXCLUDE"}
+  #   {"event_set": ["PROCESS_KPROBE"], "fields": "process", "action": "INCLUDE"}
+  #
```
Comment (kkourt) [inline install/kubernetes/values.yaml:105]: ❤️

File: api/v1/README.md [hunk 2]
```diff
@@ -793,6 +793,7 @@ AggregationOptions defines configuration options for aggregating events.
 | event_set | [EventType](#tetragon-EventType) | repeated | Event types to filter or undefined to filter over all event types. |
 | fields | [google.protobuf.FieldMask](#google-protobuf-FieldMask) |  | Fields to include or exclude. |
 | action | [FieldFilterAction](#tetragon-FieldFilterAction) |  | Whether to include or exclude fields. |
+| invert_event_set | [google.protobuf.BoolValue](#google-protobuf-BoolValue) |  | Whether or not the event set filter should be inverted. |
```
Comment (kkourt) [inline api/v1/README.md:796]: Just a question: This is not supported in the CLI, right? Is it used somewhere?

File: pkg/filters/fields_test.go [hunk 3]
```diff
@@ -272,3 +273,340 @@ func TestFieldFilterInvertedEventSet(t *testing.T) {
 	filter.Filter(ev)
 	assert.True(t, proto.Equal(ev, expected), "events are equal after filter")
 }
+
+func TestSlimExecEventsFieldFilterExample(t *testing.T) {
```
Comment (kkourt) [inline pkg/filters/fields_test.go:277]: ❤️

**Review by kkourt — APPROVED** [review]
🚀

File: api/v1/README.md (hunk 2 above)
Comment (willfindlay) [reply api/v1/README.md:796]: @kkourt right, it's not supported in the CLI at the moment, only in the configmap. I'm using it in the slim exec field filter, for example.

**Comment by jrfastab** [comment]
thanks!

**Pull request merged**